### PR TITLE
Fix check oauth email verification

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?

* In case of OAuth providers that not guarantee a verified email, we should not merge them with an existing account session with the same email 
* Remove VK OAuth provider since they no longer expose the email  

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/controllers/api/account.php [hunk 1]
```diff
@@ -476,9 +476,12 @@
         if ($user === false || $user->isEmpty()) { // No user logged in or with OAuth2 provider ID, create new one or connect with account with same email
             $name = $oauth2->getUserName($accessToken);
             $email = $oauth2->getUserEmail($accessToken);
+            $isVerified = $oauth2->isEmailVerified($accessToken);
 
-            $user = $dbForProject->findOne('users', [new Query('deleted', Query::TYPE_EQUAL, [false]), new Query('email', Query::TYPE_EQUAL, [$email])]); // Get user by email address
-
+            if ($isVerified === true) {
+                $user = $dbForProject->findOne('users', [new Query('deleted', Query::TYPE_EQUAL, [false]), new Query('email', Query::TYPE_EQUAL, [$email])]); // Get user by email address
```
Comment (eldadfux) [inline app/controllers/api/account.php:482]: We might want a new index

Comment (TorstenDittmann) [reply app/controllers/api/account.php:482]: We already have an index for this 👍🏻

File: src/Appwrite/Auth/OAuth2/Apple.php [hunk 2]
```diff
@@ -136,16 +136,31 @@ public function getUserID(string $accessToken): string
...
+     * 
+     * @link https://developer.apple.com/forums/thread/121411
+     * 
+     * @param $accessToken
+     * 
+     * @return bool
+     */
+    public function isEmailVerified(string $accessToken): bool
+    {
+        if (isset($this->claims['email_verified']) && $this->claims['email_verified'] === 'true') {
```
Comment (TorstenDittmann) [inline src/Appwrite/Auth/OAuth2/Apple.php:157]:
```suggestion
        if (($this->claims['email_verified'] ?? false) === 'true') {
```

File: src/Appwrite/Auth/OAuth2/Gitlab.php [hunk 3]
```diff
@@ -127,6 +127,26 @@ public function getUserEmail(string $accessToken): string
...
+     * 
+     * @param $accessToken
+     * 
+     * @return bool
+     */
+    public function isEmailVerified(string $accessToken): bool
+    {
+        $user = $this->getUser($accessToken);
+
+        if (isset($user['confirmed_at']) && $user['confirmed_at'] !== null) {
```
Comment (TorstenDittmann) [inline src/Appwrite/Auth/OAuth2/Gitlab.php:143]:
```suggestion
        if ($user['confirmed_at'] ?? false) {
```

File: src/Appwrite/Auth/OAuth2/Google.php [hunk 4]
```diff
@@ -137,6 +137,26 @@ public function getUserEmail(string $accessToken): string
...
+     * 
+     * @param $accessToken
+     * 
+     * @return bool
+     */
+    public function isEmailVerified(string $accessToken): bool
+    {
+        $user = $this->getUser($accessToken);
+
+        if (isset($user['email_verified']) && $user['email_verified'] === true) {
```
Comment (TorstenDittmann) [inline src/Appwrite/Auth/OAuth2/Google.php:153]:
```suggestion
        if ($user['email_verified'] ?? false) {
```

**Review by TorstenDittmann — CHANGES_REQUESTED** [review]
lgtm, but we can improve some code quality by removing all the `isset` calls.

**Pull request merged**